### PR TITLE
Initial Stylekit documentation

### PR DESCRIPTION
**Pull request by FabienLelaquais**
Addresses #398.

### CODE REVIEW

File: docs/manuals/gui/configuration.md [hunk 1]
```diff
@@ -72,6 +72,19 @@ Here is the list of the configuration parameters you can use in
...
      Or you can force **not** to use the *debug* mode using the *--no_debug* option
      when launching your application.
+   - *theme* (Union[dict[str, any], None]): A dictionary that lets you customize
+     the theme of your application. See the [Themes section](styling.md/#themes) for
+     details.
+   - *light_theme* (Union[dict[str, any], None]): Similar to the *theme* setting,
+     but applies to the _light_ theme only.
+   - *dark_theme* (Union[dict[str, any], None]): Similar to the *theme* setting,
+     but applies to the *dark* theme only.
+   - *stylekit* (Union[bool, dict[str, any], None]): If True or unspecified, use the
```
Comment (FredLL-Avaiga) [inline docs/manuals/gui/configuration.md]: Should any be union[string,int] ?

File: docs/manuals/gui/styling/index.md [hunk 2]
```diff
@@ -7,18 +7,23 @@ some style to, providing the best user experience.
 Styling involves some knowledge of [Cascading Style Sheets](https://www.w3.org/Style/CSS/).
 This section describes what sort of styling you may want to apply, in different situations.
 
+Taipy GUI comes with the [Stylekit](stylekit.md): a set of stylesheets that provide a default,
+homogeneous, easy to customize style for all visual elements and a few utility CSS classes that
+accelerates the development of specific CSS classes to apply to pages. The Stylekit is installed
```
Comment (FredLL-Avaiga) [inline docs/manuals/gui/styling/index.md]: Rephrase?

File: docs/manuals/gui/styling/stylekit.md [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+
+Taipy GUI comes with predefined cascading stylesheets that solidify your application's
+general look and feel. Those stylesheets also make it far easier to customize styles.</br>
+All those stylesheets are grouped in what is called the *Stylekit*.
+
+The Stylekit provides a few useful features:
+
+- Predefined styles for Taipy GUI visual elements so they align better on the page;
+- CSS variables that are used all over the stylesheets so that you can modify those
+  variables and witness the propagation of specific graphical attributes;
```
Comment (FredLL-Avaiga) [inline docs/manuals/gui/styling/stylekit.md]: Mainly colors

File: docs/manuals/gui/styling/stylekit.md [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+The exhaustive list of all the Stylekit variables can be found in the
+[Variables](#variables) section below.
+
+## Variables
+
+The Stylekit defines several variables that impact the overall style of your Taipy GUI
+application pages. The values for these variables can be overloaded by providing a dictionary
+to the *stylekit* parameter of the `Gui.run()^` method.
+
+Each variable initializes a CSS custom property that you can use in your own stylesheets.<br/>
```
Comment (FredLL-Avaiga) [inline docs/manuals/gui/styling/stylekit.md:60]: And modify the Mui base theme

**Review by FredLL-Avaiga — APPROVED** [review]

File: docs/manuals/gui/configuration.md (hunk 1 above)
Comment (FabienLelaquais) [reply docs/manuals/gui/configuration.md]: True. I was wondering if that didn't make the reading harder.
And if we add something like opacity, then 'number' would come into the game as well...

What do you think?

File: docs/manuals/gui/styling/stylekit.md (hunk 4 above)
Comment (FabienLelaquais) [reply docs/manuals/gui/styling/stylekit.md:60]: Although this is true, I'm trying to hide those implementation details from 'regular' users.
May be I could create a more technical document in taipy-gui/gui to expose those kind of things.
Thoughts?

File: docs/manuals/gui/styling/index.md (hunk 2 above)
Comment (FabienLelaquais) [reply docs/manuals/gui/styling/index.md]: Do you prefer:

"Taipy GUI comes with a set of stylesheets that give applications an homogeneous and solid
look and feel. This set of stylesheet is grouped in what is called the Stylekit.
The Stylekit is easy to customize and provides few utility CSS classes that accelerates the
development of specific styles to apply to pages.
The Stylekit is documented in the Stylekit section of this documentation, where you can find
all the information on how to use and customize it.
"
?

**Comment by FredLL-Avaiga** [comment]
🫶

File: docs/manuals/gui/styling/index.md (hunk 2 above)
Comment (ghost) [reply docs/manuals/gui/styling/index.md]: With some corrections:

> Taipy GUI comes with a set of stylesheets that give applications a homogeneous and solid
look and feel. This set of stylesheets is grouped in what is called the Stylekit.
The Stylekit is easy to customize and provides a few CSS utility classes that accelerate the
development of specific styles to apply to pages.
The Stylekit is documented in the Stylekit section of this documentation, where you can find
all the information on how to use and customize it.

Maybe on that last line, put a link on the words « Stylekit section » to the actual section.
But besides that, it seems fine to me.

File: docs/manuals/gui/styling/index.md [hunk 5]
```diff
@@ -43,29 +48,59 @@ font size, you could write:
...
+A text that has a **specific**{: .blue } style.
+```
+will apply the *blue* class to the bold text fragment:
+<figure>
+  <img src="attribute-fragment-d.png" class="visible-dark" />
+  <img src="attribute-fragment-l.png" class="visible-light"/>
+  <figcaption>Styling fragments</figcaption>
+</figure>
+
+Note to style a heading, the syntax is slightly different: right after the text of the heading,
```
Comment (ghost) [inline docs/manuals/gui/styling/index.md]: > Please note that, to style a heading, the syntax...

File: docs/manuals/gui/styling/index.md [hunk 6]
```diff
@@ -43,29 +48,59 @@ font size, you could write:
...
+you must insert a white space, as many sharp (#) characters as the heading uses, another white
+space, then the style construct.<br/>
```
Comment (ghost) [inline docs/manuals/gui/styling/index.md:99]: "then the _Markdown Attribute Lists_ construct" would be more exact, but if that’s fine by you, it is by me as well.

File: docs/manuals/gui/configuration.md (hunk 1 above)
Comment (ghost) [reply docs/manuals/gui/configuration.md]: That’s right… CSS in JS values can be `string`, `number`, or `int`. I don’t recall any other type, though. This seems pretty extensive to me.
Can’t you specify those three types?

**Review by ghost — APPROVED** [review]
👍

File: docs/manuals/gui/configuration.md (hunk 1 above)
Comment (FredLL-Avaiga) [reply docs/manuals/gui/configuration.md]: then it should be [string, int, float]

File: docs/manuals/gui/styling/stylekit.md (hunk 4 above)
Comment (FredLL-Avaiga) [reply docs/manuals/gui/styling/stylekit.md:60]: that's a plan

File: docs/manuals/gui/styling/index.md (hunk 2 above)
Comment (FabienLelaquais) [reply docs/manuals/gui/styling/index.md]: The link is there already...

File: docs/manuals/gui/configuration.md (hunk 1 above)
Comment (FabienLelaquais) [reply docs/manuals/gui/configuration.md]: I will document this as [string, int, float]

File: docs/manuals/gui/styling/index.md (hunk 5 above)
Comment (FabienLelaquais) [reply docs/manuals/gui/styling/index.md]: 👍

**Pull request merged**